app.js

```diff
@@ -13,69 +13,89 @@ const teamOneLabel = document.getElementById('team-one-name');
 const teamTwoLabel = document.getElementById('team-two-name');
 
 // create an array to hold on to the state of past games
-
-let name1 = '',
-let name2 =  '',
-let score1 = 0,
-let score2 = 0,
+const pastGamesArray = [];
+let name1 = '';
+let name2 = '';
+let score1 = 0;
+let score2 = 0;
 
 nameForm.addEventListener('submit', (e) => {
     // don't forget to prevent the default form behavior!
-
+    e.preventDefault();
     // get the name data from the form
-
+    const data = new FormData(nameForm);
     // set the state to this data from the form
-
+    name1 = data.get('name-1');
+    name2 = data.get('name-2');
     // reset the form values
+    teamOneLabel.textContent = name1;
+    teamTwoLabel.textContent = name2;
 
     displayCurrentGameEl();
 });
 
 
 teamOneAddButton.addEventListener('click', () => {
     // increment the current state for team one's score
-    
+    score1++;
+
     displayCurrentGameEl();});
 
 teamTwoAddButton.addEventListener('click', () => {
     // increment the current state for team two's score
-
+    score2++;
     displayCurrentGameEl();
 });
 
 teamOneSubtractButton.addEventListener('click', () => {
     // decrement the current state for team one's score
-
+    score1--;
     displayCurrentGameEl();
 });
 
 teamTwoSubtractButton.addEventListener('click', () => {
     // decrement the current state for team two's score
-
+    score2--;
     displayCurrentGameEl();
 });
 
 finishGameButton.addEventListener('click', () => {
-    
+    nameForm.reset();
     // add the current game to an array of games in state.
     // HINT: it will be helpful to keep track of these games as objects with 4 properties, one for each piece of state we're tracking
+    const game = {
+        name1: name1,
+        name2: name2,
+        score1: score1,
+        score2: score2,
+    };
+    
+    pastGamesArray.push(game);
 
     displayAllGames();
-
+    
     // reset the initial state to start with a new form
+    teamOneLabel.textContent = '';
+    teamTwoLabel.textContent = '';
 
     displayCurrentGameEl();
 });
 
 
 function displayCurrentGameEl() {
     // clear out the current game div
-
+    pastGamesEl.textContent = '';
     // change the label to show team one's name;
+    teamOneLabel.textContent = name1;
     // change the label to show team two's name;
-
+    teamTwoLabel.textContent = name2;
     // call the render game function to create a game element
-    
+
+    for (let pastGame of pastGamesArray) {
+        const container = renderGame(pastGame);
+
+        pastGamesEl.append(container);
+    }
```

Comment:
Reading the name of a function can (if it is named well) give you a hint as to what should be happening inside the function declaration.  You are attempting to write `displayCurrentGameEL`.  On line 94, you are looping through and trying to display all past games.  Look at the `renderGame` function in your utils file.  It takes one arguement, a game object.  Where have you created a game object already in your code? (Look at lines 66-71).  Instead of using a for loop here, I would create another game object like you did on line 66 and pass this as an argument when you invoke `renderGame`.  `renderGame` should return the game DOM element that you need here, which you can append to the game div below.